File: collections/vector-fill.kt
import java.time.Instant
import java.time.temporal.ChronoUnit

data class Data(val a: Long, val b: Long)

fun newData(x: Int): Data {
    return Data(x.toLong() / 2, x.toLong() / (x % 2 + 1).toLong())
}

fun main() {
    val startTime = Instant.now()
    //
    val vec = ArrayList<Data>(100_000)
    for (i in 0 until 100_000) {
        vec.add(newData(i))
    }
    //
    val endTime = Instant.now()
    val elapsedMicros = ChronoUnit.MICROS.between(startTime, endTime)
    println("$elapsedMicros")
}

File: collections/vector-fill.rs
#![allow(dead_code)]

use std::time::Instant;

struct Data {
    a: u64,
    b: u64,
}

impl Data {
    fn new(x: usize) -> Data {
        Data {
            a: x as u64 / 2,
            b: x as u64 / (x % 2 + 1) as u64,
        }
    }
}

fn main() {
    let start_time = Instant::now();
    //
    {
        let mut vec: Vec<Data> = Vec::with_capacity(100_000);
        for i in 0..100_000 {
            vec.push(Data::new(i));
        }
    }
    //
    let end_time = Instant::now();
    let elapsed_time = end_time.duration_since(start_time).as_micros();
    println!("{}", elapsed_time);
}
